miio/walkingpad.py

```diff
@@ -0,0 +1,189 @@
+    # raw_command get_prop '["sensitivity"]' returns [2]
+    # raw_command get_prop '["sp"]' returns [1.0]
+    # raw_command get_prop '["start_speed"]' returns [1.0]
+    # raw_command get_prop '["auto"]' returns [1]
+    # raw_command get_prop '["disp"]' returns [19]
+    # raw_command get_prop '["lock"]' returns [0]
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+
+        # NOTE: Only 1 property can be requested at the same time
+        self.data = data
+
+    @property
+    def power(self) -> str:
+        """Power state."""
+        return self.data["power"]
+
+    @property
+    def is_on(self) -> bool:
+        """True if the device is turned on."""
+        return self.power == "on"
+
+    @property
+    def time(self) -> int:
+        """Current walking time."""
+        return self.data["time"]
+
+    @property
+    def speed(self) -> float:
+        """Current speed."""
+        return self.data["sp"]
+
+    @property
+    def mode(self) -> int:
+        """Current mode."""
+        return self.data["mode"]
+
+    @property
+    def step_count(self) -> int:
+        """Current steps."""
+        return self.data["step"]
+
+    @property
+    def distance(self) -> int:
+        """Current distance."""
+        return self.data["dist"]
+
+    @property
+    def calories(self) -> int:
+        """Current distance."""
+        return self.data["cal"]
+
+
+class Walkingpad(Device):
+    """Main class representing Xiaomi Walkingpad."""
+
+    # TODO: - Auto On/Off Not Supported
+    #       - Adjust Scenes with Wall Switch Not Supported
+
+    @command(
+        default_output=format_output(
+            "",
+            "Mode: {result.mode}\n"
+            "Time: {result.time}\n"
+            "Steps: {result.step}\n"
+            "Speed: {result.speed}\n"
+            "Distance: {result.distance}\n"
+            "Calories: {result.calories}  ",
+        )
+    )
+    def status(self) -> WalkingpadStatus:
+        """Retrieve properties."""
+
+        # Walkingpad A1 allows you to retrieve a subset of values with "all"
+        # eg ['mode:1', 'time:1387', 'sp:3.0', 'dist:1150', 'cal:71710', 'step:2117']
+
+        properties = ["all"]
+
+        # Walkingpad A1 only allows 1 property to be read at a time
+
+        values = self.get_properties(properties, max_properties=1)
+
+        # When running the tests, for some reason the list provided is passed within another list, so I take
+        # care of this here.
+
+        if any(isinstance(el, list) for el in values):
+            values = values[0]
+
+        data = {}
+        for x in values:
+            prop, value = x.split(":")
+            data[prop] = value
+
+        properties_additional = ["power", "mode"]
+        values_additional = self.get_properties(properties_additional, max_properties=1)
+
+        for i in range(len(properties_additional)):
+            data[properties_additional[i]] = values_additional[i]
```

Comment:
```suggestion
additional_props = dict(zip(properties_additional, values_additional))
data.update(additional_props)
```
would be more pythonic.

Reply:
Nice! I have changed to this 👍